posthog/async_migrations/runner.py

```diff
@@ -77,23 +79,32 @@ def start_async_migration(migration_name: str, ignore_posthog_version=False) ->
 
     mark_async_migration_as_running(migration_instance)
 
-    return run_async_migration_next_op(migration_name, migration_instance)
+    return run_async_migration_operations(migration_name, migration_instance)
+
 
+def run_async_migration_operations(migration_name: str, migration_instance: Optional[AsyncMigration] = None) -> bool:
+    while True:
+        run_next, success = run_async_migration_next_op(migration_name, migration_instance)
+        if not run_next:
+            return success
```

Comment:
👍  adding this function
Consider (so `run _next_op` could just return success/failure which would be easier to read imho)
```suggestion
    while (not migration_instance or migration_instance.status !=  MigrationStatus.CompletedSuccessfully):
        if not run_async_migration_next_op(migration_name, migration_instance):
          return False
    return True
```

Reply:
The proposed refactoring does not work for handling failures. Will think on the idea tomorrow.

Reply:
If there's a failure we'd `return False` (if `run_async_migration_next_op` failed, it doesn't return `run_next` only `success` based on my proposal) on line 88 in my proposal, or did I misunderstand what you meant by handling failures?

Reply:
I think the original here is very easy to read if you want a second opinion here